frontend/plan/src/components/selector/CourseList.js

```diff
@@ -31,37 +31,76 @@ const courseSort = (courses, sortMode) => {
     return sorted;
 };
 
-export default function CourseList({ courses, getCourse, sortMode }) {
-    return (
-        <div className="scroll-container">
-            <div style={{
-                display: "flex", flexDirection: "row", paddingBottom: "1em", paddingLeft: "2em",
-            }}
-            >
-                <div className="header" style={{ overflow: "hidden", width: "60%" }}>
+export default class CourseList extends React.Component {
+    constructor(props) {
+        super(props);
+        this.listRef = React.createRef();
+    }
+
+
+    componentDidMount() {
+    // If we have a snapshot value, we've just added new items.
+    // Adjust scroll so these new items don't push the old ones out of view.
+    // (snapshot here is the value returned from getSnapshotBeforeUpdate)
+        const { scrollPos } = this.props;
+        const list = this.listRef.current;
+        list.scrollTop = scrollPos;
+    }
+
+    componentWillUnmount() {
```

Comment:
Just a comment that doesn't require a rewrite - componentWillUnmount can be emulated with hooks by placing this code in a function in the return statement of useEffect.

Reply:
moved back to functional component using the useEffect cleanup functionality!